src/Datadog.Trace/Configuration/ConfigurationKeys.cs

```diff
@@ -219,6 +233,13 @@ public static class ConfigurationKeys
         /// </summary>
         public const string DiagnosticSourceEnabled = "DD_DIAGNOSTIC_SOURCE_ENABLED";
 
+        /// <summary>
+        /// Configuration key for overriding the transport to use for communicating with the trace agent.
+        /// Default value is <c>null</c>.
+        /// Options available: datadog-http, windows-named-pipes
```

Comment:
```suggestion
        /// Valid values: <c>datadog-http</c>, <c>windows-named-pipes</c>
```

Reply:
First thought: `datadog-http` seems confusing since both options use http. I think we can also drop "Windows" from all the named pipes references.

Second thought: Sounds like this setting determines which http client to use and the underlying transport. There are currently 4 combinations, right?

- `System.Net.Http.HttpClient` (default on .NET Core 3.1, always TCP)
- `System.Net.HttpWebRequest` (default on .NET Framework, always TCP)
- `Datadog...HttpClient` with TCP (should we allow this?)
- `Datadog...HttpClient` with named pipes

Reply:
I don't think the user should be able to choose between `HttpClient` and `WebRequest`.
I like the idea of keeping the custom TCP implementation in, at least as an internal undocumented option so we can work to battle-harden it in the reliability environment.
Happy to rename `datadog-http` to `datadog-tcp`.
My thought was to keep windows in these variable names to avoid conflicts in the future with named pipes in linux as the implementation may differ, but maybe there will be no difference?

Reply:
Yeah, I agree we don't need to provide `HttpClient` and `WebRequest` now. We may allow that in the future, though, so we should pick names that don't make it awkward later.

About "Windows" in the name, it looks like we would use `NamedPipeClientStream` regardless of the OS?

What do you think of `datadog-tcp` and `datadog-named-pipes` since they both use the Datadog http client? Later if we wanted to we can add something like `microsoft-httpclient` or `system-httpclient` for the others.